### PR TITLE
Catching of "No such host is known" exception

### PR DESCRIPTION
**Pull request by anhr**
I have added the catching of exception if server address is invalid or
no internet connection. Also I have added the IRCClient console
application for testing.

### CODE REVIEW

File: ChatSharp/Events/ErrorEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+    /// <summary>
+    /// Raised when a Error occurs.
+    /// </summary>
+    public class ErrorEventArgs : EventArgs
+    {
+        /// <summary>
+        /// The error that has occured.
+        /// </summary>
+        public string Error { get; set; }
```
Comment (SirCmpwn) [inline ChatSharp/Events/ErrorEventArgs.cs]: I would rather include the actual Exception than e.Message

**Comment by SirCmpwn** [comment]
Please remove the IRCClient project from this PR.

File: ChatSharp/Events/ErrorEventArgs.cs (hunk 1 above)
Comment (anhr) [inline ChatSharp/Events/ErrorEventArgs.cs]: Ready. See https://github.com/anhr/ChatSharp/commit/d11905962f4af4b92668a09d21a16c7daf4e4913

**Comment by SirCmpwn** [comment]
Sorry I didn't think of this earlier - how about also including a boolean that tells if the error is fatal, and if so, clean up and prepare the client to reconnect?

**Comment by SirCmpwn** [comment]
We could also just mandate that after fatal errors the client must be Disposed and not used again.

**Comment by anhr** [comment]
Actual Exception instead e.Message. Removed the IRCClient project from this PR. See https://github.com/anhr/ChatSharp/commit/d11905962f4af4b92668a09d21a16c7daf4e4913

**Comment by anhr** [comment]
I am not ready to answer to your question because I do not full clear in your project and I need to research of your project.

**Comment by SirCmpwn** [comment]
No problem, take your time.

**Comment by anhr** [comment]
IRC server error replies handler https://github.com/anhr/ChatSharp/commit/4a498fc429a4ca7c85e1895f7829a03907a5a8ce

**Comment by SirCmpwn** [comment]
The singular form of "replies" is "reply".

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**